### PR TITLE
Update CI

### PR DESCRIPTION
**Pull request by dolfinus**
Several small updates of CI workflow. Changes are described below.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,60 +1,127 @@
...
       - master
 
 jobs:
   mix_test:
     name: mix test (OTP ${{matrix.otp}} | Elixir ${{matrix.elixir}})
+
     strategy:
       matrix:
         include:
-          - otp: 21.3.8.16
```
Comment (dolfinus) [inline .github/workflows/ci.yml:15]: It is hard to read this matrix without line breaks.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,60 +1,127 @@
 name: CI
 
 on:
-  pull_request:
   push:
```
Comment (dolfinus) [inline .github/workflows/ci.yml:4]: Running tests only on master branch does not seem right.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -1,60 +1,127 @@
...
+      uses: actions/checkout@v2
+
+    - name: Set up Elixir
+      uses: erlef/setup-elixir@v1
       with:
-        otp-version: ${{matrix.otp}}
-        elixir-version: ${{matrix.elixir}}
-    - name: Install Dependencies
-      run: |
-        mix local.rebar --force
```
Comment (dolfinus) [inline .github/workflows/ci.yml:34]: rebar and hex are updated in `erlef/setup-elixir` action itself, no need to do that again.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -1,60 +1,127 @@
...
+          - elixir: 1.9.4
+            otp: 21.3.8.16
+
+          - elixir: 1.10.4
+            otp: 21.3.8.16
+
+          - elixir: 1.10.4
+            otp: 23.0.2
+
+          - elixir: 1.11.4
```
Comment (dolfinus) [inline .github/workflows/ci.yml:31]: Add latest Elixir and OTP releases into the matrix

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -1,60 +1,127 @@
...
-    - name: Install Dependencies
-      run: |
-        mix local.rebar --force
-        mix local.hex --force
-        mix deps.get
-    - name: Run Tests
+        elixir-version: ${{ matrix.elixir }}
+        otp-version: ${{ matrix.otp }}
+
+    - name: Restore deps and _build cache
```
Comment (dolfinus) [inline .github/workflows/ci.yml:47]: Cache dependencies to speed up workflow run

File: lib/phoenix_live_view/diff.ex [hunk 6]
```diff
@@ -230,16 +230,15 @@ defmodule Phoenix.LiveView.Diff do
 
     case cid_to_component do
       %{^cid => {component, _id, assigns, private, fingerprints}} ->
-          socket
```
Comment (dolfinus) [inline lib/phoenix_live_view/diff.ex:233]: Here and below source code format was fixed.

File: mix.lock [hunk 7]
```diff
@@ -1,14 +1,9 @@
 %{
-  "earmark_parser": {:hex, :earmark_parser, "1.4.12", "b245e875ec0a311a342320da0551da407d9d2b65d98f7a9597ae078615af3449", [:mix], [], "hexpm", "711e2cc4d64abb7d566d43f54b78f7dc129308a63bc103fbd88550d2174b3160"},
```
Comment (dolfinus) [inline mix.lock:2]: Unused dependencies.

**Comment by josevalim** [comment]
Please don’t run the formatter on this project yet, because there are many files that were never formatted before and we would have to do so manually. Thanks! :)

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -1,60 +1,127 @@
...
+          deps
+          _build
+        key: deps-${{ runner.os }}-${{ matrix.otp }}-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}
+        restore-keys: |
+          deps-${{ runner.os }}-${{ matrix.otp }}-${{ matrix.elixir }}
+
+    - name: Install dependencies
+      run: mix deps.get --only test
+
+    #- name: Check source code format
```
Comment (dolfinus) [inline .github/workflows/ci.yml:60]: Please run `mix format` before uncommenting these lines

**Comment by dolfinus** [comment]
Ok, reverted that change.

**Comment by dolfinus** [comment]
Hello.

Can anyone review this PR, please?

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by dolfinus** [comment]
Thanks!

**Pull request merged**